packages/wallet/src/W3mFrameSchema.ts

```diff
@@ -310,6 +310,48 @@ export const RpcEthSendTransactionRequest = z.object({
   params: z.array(z.any())
 })
 
+export const RpcSolanaSignMessageRequest = z.object({
+  method: z.literal('solana_signMessage'),
+  params: z.object({
+    message: z.string(),
+    pubkey: z.string()
+  })
+})
+
+export const RpcSolanaSignTransactionRequest = z.object({
+  method: z.literal('solana_signTransaction'),
+  params: z.object({
+    transaction: z.string()
+  })
+})
+
+export const RpcSolanaSignAndSendTransactionRequest = z.object({
+  method: z.literal('solana_signAndSendTransaction'),
+  params: z.object({
+    transaction: z.string(),
+    // Options should match https://solana-labs.github.io/solana-web3.js/types/SendOptions.html
```

Comment:
any reason to base it off solana-web3js specfically?

Reply:
It is what exposes the exact options for sending a transaction to Solana and what is commonly used for interact with Solana. Do you think we should change it?